serde-avro/src/de/mod.rs

```diff
@@ -65,20 +66,24 @@ impl<'a, R> Deserializer<'a, R>
     where R: io::Read + read::Limit
 {
     pub fn new(input: R,
+               header: &'a header::Header,
```

Comment:
This is a breaking API change, just to highlight that!

Reply:
Are you opposed to having a new2 and a new_cow2?

I figure this is safest to ensure we don't break the API.